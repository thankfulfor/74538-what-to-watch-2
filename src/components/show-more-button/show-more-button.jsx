import * as React from 'react';
import PropTypes from 'prop-types';

export const ShowMoreButton = (props) => {
  const {onClick} = props;

  return (
    <div className="catalog__more">
      <button onClick={onClick} className="catalog__button" type="button">
        Show more
      </button>
    </div>
  );
};

ShowMoreButton.propTypes = {
  onClick: PropTypes.func.isRequired
};
