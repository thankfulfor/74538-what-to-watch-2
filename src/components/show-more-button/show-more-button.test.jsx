import React from 'react';
import {ShowMoreButton} from './show-more-button.jsx';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({adapter: new Adapter()});

it(`ShowMoreButton rendered`, () => {
  const clickHandler = jest.fn();
  const historyMock = {push: jest.fn()};
  const tree = shallow(<ShowMoreButton history={historyMock} onClick={clickHandler} />);

  expect(tree).toMatchSnapshot();
});
