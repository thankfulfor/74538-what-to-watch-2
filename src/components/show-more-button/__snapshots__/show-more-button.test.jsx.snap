// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowMoreButton rendered 1`] = `ShallowWrapper {}`;
